### PR TITLE
Upgrade tsickle to TypeScript 2.6

### PR DESCRIPTION
**Pull request by mprobst**
See the individual commits for more details. The major bit here is how we handle imports and exports.

### CODE REVIEW

**Comment by mprobst** [comment]
Admittedly the `options` and `host` fields in the flags are named suboptimal ;-) Taking ideas, maybe I should just say `compilerOptions` and `moduleResolutionHost`? OTOH this actually runs out to line up with the fields that `CompilerHost` in rules_typescript already defines.

File: src/typescript.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import * as ts from 'typescript';
 // Note, this import depends on a genrule copying the .d.ts file to this package
 import * as ts24 from './typescript-2.4';
```
Comment (alexeagle) [inline src/typescript.ts]: I think we now need to add a test case for ts25 as well

File: src/es5processor.ts [hunk 2]
```diff
@@ -119,6 +149,7 @@ class ES5Processor extends Rewriter {
 
   /**
    * Emits file comments for the current source file, if any.
+   * oddityasds
```
Comment (rkirov) [inline src/es5processor.ts]: oddity indeed

File: src/transformer_sourcemap.ts [hunk 3]
```diff
@@ -49,10 +49,33 @@ export function createTransformerFromSourceMap(
...
+        if (ts.isExportDeclaration(node)) {
+          // Return the original nodes for export declarations, unless they were expanded from an
+          // export * to specific exported symbols.
+          const originalExport = originalNode as ts.ExportDeclaration;
+          if (!node.moduleSpecifier) {
+            // export {a, b, c};
+            return originalNode;
+          }
+          if (!!originalExport.exportClause === !!node.exportClause) {
+            // This alread was export with symbols (export {...}) or was not expanded.
```
Comment (rkirov) [inline src/transformer_sourcemap.ts]: already

File: src/typescript.ts (hunk 1 above)
Comment (mprobst) [reply src/typescript.ts]: Done.

File: src/transformer_sourcemap.ts (hunk 3 above)
Comment (mprobst) [reply src/transformer_sourcemap.ts]: Done.

File: src/es5processor.ts (hunk 2 above)
Comment (mprobst) [reply src/es5processor.ts]: Fixed.

**Pull request merged**